addon/PlacesProvider.js

```diff
@@ -684,6 +676,20 @@ Links.prototype = {
   },
 
   /**
+   * Process links after getting them from the database.
+   *
+   * @param {Array} links
+   *            an array containing link objects
+   *
+   * @returns {Array} an array of checked links with favicons and eTLDs added
+   */
+  _processLinks(links) {
+    return this._addETLD(
+      this._faviconBytesToDataURI(links)
+    ).filter(link => LinkChecker.checkLoadURI(link.url));
```

Comment:
This is an odd combination of nesting and chaining, so keeping the original sequential style would be easier to parse. The ordering doesn't quite matter too much right now (although I suppose the filter could remove links first to not even bother with favicon or eTLD), and the sequential statements would keep that clear.